apps/onnx/CMakeLists.txt

```diff
@@ -0,0 +1,162 @@
+    PROTOS ${ONNX_PROTO_FILE}
+    IMPORT_DIRS ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+# Add library that converts ONNX models to Halide operators
+add_library(oclib STATIC onnx_converter.cc ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx.pb.cc)
+target_include_directories(
+    oclib
+    PUBLIC
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${Protobuf_INCLUDE_DIRS}
+)
+target_link_libraries(oclib PUBLIC Halide::Halide ${Protobuf_LITE_LIBRARIES})
+target_compile_definitions(oclib PUBLIC GOOGLE_PROTOBUF_NO_RTTI)
+
+add_executable(onnx_converter_test onnx_converter_test.cc)
+target_link_libraries(onnx_converter_test oclib)
+
+add_test(NAME onnx_converter_test COMMAND onnx_converter_test)
+set_tests_properties(
+    onnx_converter_test PROPERTIES
+    LABELS onnx
+    PASS_REGULAR_EXPRESSION "Success!"
+    SKIP_REGULAR_EXPRESSION "\\[SKIP\\]"
+)
+
+# Generator
+add_halide_generator(
+    onnx_converter.generator
+    SOURCES onnx_converter_generator.cc
+    LINK_LIBRARIES oclib
+)
+
+# Generate test onnx model
+add_custom_command(
+    OUTPUT test_model.onnx
+    COMMAND
+    ${CMAKE_COMMAND} -E cat ${CMAKE_CURRENT_SOURCE_DIR}/test_model_proto.txt
+    | protoc --encode=onnx.ModelProto --proto_path=${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/onnx.proto
+    > test_model.onnx
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/onnx.proto
+    COMMENT "Generating test ONNX model from proto content"
+    MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/test_model_proto.txt
+    VERBATIM
+)
+add_custom_target(test_model_onnx DEPENDS test_model.onnx)
+add_dependencies(test_model_onnx onnx_proto)
+
+# Generate static library using halide generator for test onnx model
+add_halide_library(
+    test_model FROM onnx_converter.generator
+    GENERATOR onnx_model_generator
+    PARAMS model_file_path=${CMAKE_CURRENT_BINARY_DIR}/test_model.onnx
+    AUTOSCHEDULER Halide::Adams2019
+)
+add_dependencies(test_model test_model_onnx)
+
+# Test the generated static library
+add_executable(onnx_converter_generator_test onnx_converter_generator_test.cc)
+target_link_libraries(onnx_converter_generator_test PRIVATE Halide::Runtime test_model)
+
+add_test(NAME onnx_converter_generator_test COMMAND onnx_converter_generator_test)
+set_tests_properties(
+    onnx_converter_generator_test
+    PROPERTIES
+    LABELS onnx
+    PASS_REGULAR_EXPRESSION "Success!"
+    SKIP_REGULAR_EXPRESSION "\\[SKIP\\]"
+)
+
+# Python bindings to convert onnx models to Halide model
+find_package(Python3 COMPONENTS Interpreter Development)
+if (NOT Python3_FOUND)
+    message(WARNING "Could NOT find Python3")
+    return()
+endif ()
+
+find_package(pybind11)
+if (NOT pybind11_FOUND)
+    message(WARNING "Could NOT find pybind11")
+    return()
+endif ()
+
+pybind11_add_module(model_cpp model.cpp)
+target_link_libraries(model_cpp PRIVATE Halide::Halide oclib)
+if (UNIX)
+    target_link_libraries(model_cpp PRIVATE -Wl,--no-as-needed Halide::Adams2019 -Wl,--as-needed)
+endif ()
```

Comment:
I doubt this will work with the Apple linker... in `Makefile.inc`, @abadams uses `-force_load`, but I doubt that will work for a loadable module (`.so`) these days. Does the Makefile build the autoschedulers as static libraries ever?